transformer/transformers/ampruntimejs.go

```diff
@@ -0,0 +1,35 @@
+package transformers
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/ampproject/amppackager/transformer/internal/amphtml"
+	"github.com/ampproject/amppackager/transformer/internal/htmlnode"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+)
+
+// AMPRuntimeJS rewrites the value of src in script nodes, where applicable.
+// If the value is of the form "*.js", replace it with "*.js?f=sxg".
+func AMPRuntimeJS(e *Context) error {
+	for n := e.DOM.HeadNode; n != nil; n = htmlnode.Next(n) {
```

Comment:
Close, we don't want to use `htmlnode.Next` anymore, since that will actually traverse out of the head and into the body.

```suggestion
	for n := e.DOM.HeadNode; n != nil; n = n.NextSibling {
```

Reply:
Done.

Reply:
Actually, doing `htmlnode.Next` was correct. Replacing with `n.NextSibling` actually skipped over the transformation altogether. I wrote a quick test that transformed two scripts, and `htmlnode.Next` passes both.